src/application.rs

```diff
@@ -106,6 +113,7 @@ pub async fn run(
           .cookie_name(HEADER_TOKEN.to_string())
           .build(),
       )
+      .wrap(MetricsMiddleware)
```

Comment:
Maybe move to line 110 (under the IdentityMiddleware). IIRC, the middleware are invoked as reverser order.